mcstatus/tests/test_retry_decorator.py

```diff
@@ -19,15 +19,15 @@ def func():
 
 
 def test_sync_fail():
-    x = -1
+    already_ran = False
 
     @retry(tries=2)
     def func():
-        nonlocal x
-        x += 1
-        if x == 0:
+        nonlocal already_ran
+        if already_ran:
+            already_ran = True
             raise OSError("First error")
-        elif x == 1:
+        else:
             raise RuntimeError("Second error")
```

Comment:
Using a boolean flag isn't enough here, since we need to explicitly test that the code ran exactly twice, not just that it ran more than once. If the retry re-run the code 5 times, even though we set `retry=2`, this test should fail, which would originally happen because once `x` gets over 1, there is no if covering that and the function runs without errors, hence failing the pytest.raises assertion.

Reply:
Coverage report showed that this code was ran only 2 times, so if someone will change it to more than 2 times - they will need to change tests, as it's with all changes.

Reply:
? I meant the if `retry` decorator implementation is badly implemented and reruns the code more times than specified, a unit-test should catch it. The original behavior does catch this, the new one does not.

Reply:
Reverted in [the last force-push](https://github.com/py-mine/mcstatus/compare/20cad3b7031cbef5dd4ccc91d0eaf6987a4f6682..b9cabb9b6da2c192cc40fd6e96ac843d569d8e64).